test/e2e/xsite/xsite_test.go

```diff
@@ -564,3 +628,22 @@ func expectsCrossSiteService(testkube *crossSiteKubernetes) {
 	}, &corev1.Service{})
 	tutils.ExpectNoError(err)
 }
+
+func expectGossipRouterSupectValue(t *testing.T, siteKube *crossSiteKubernetes, expectValue bool) {
+	deployment := siteKube.kube.GetDeployment(siteKube.crossSite.GetGossipRouterDeploymentName(), siteKube.namespace)
```

Comment:
The `GetDeployment` method excepts that the Deployment might not be found, in which case you will get an array oob error on the following line. I think the Deployment must exist.

We already have the `WaitForDeployment` and `WaitForDeploymentState` methods, so I don't think there's a need to add `GetDeployment`.